pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -0,0 +1,98 @@
+package machinehealthcheck
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const (
+	CONFIG_NAMESPACE string = "aro.machinehealthcheck"
+	ENABLED          string = CONFIG_NAMESPACE + ".enabled"
+)
+
+type Reconciler struct {
+	log *logrus.Entry
+
+	arocli aroclient.Interface
+	dh     dynamichelper.Interface
+}
+
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, dh dynamichelper.Interface) *Reconciler {
+	return &Reconciler{
+		log:    log,
+		arocli: arocli,
+		dh:     dh,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
Try and add unit test coverage for all of these returns, if it's possible. I use the following to test coverage:
```
        t="/tmp/go-cover.$$.tmp" 
        go test -coverprofile=$t $@ && go tool cover -html=$t && unlink $t
```